lib/features/email/data/network/email_api.dart

```diff
@@ -551,37 +554,25 @@ class EmailAPI with HandleSetErrorMixin {
     EmailId oldEmailId,
     {CancelToken? cancelToken}
   ) async {
-    final idCreateMethod = Id(_uuid.v1());
-    final setEmailMethod = SetEmailMethod(accountId)
-      ..addCreate(idCreateMethod, newEmail)
-      ..addDestroy({oldEmailId.id});
-
-    final requestBuilder = JmapRequestBuilder(_httpClient, ProcessingInvocation());
-
-    final setEmailInvocation = requestBuilder.invocation(setEmailMethod);
-
-    final capabilities = setEmailMethod.requiredCapabilities
-      .toCapabilitiesSupportTeamMailboxes(session, accountId);
-
-    final response = await (requestBuilder
-        ..usings(capabilities))
-      .build()
-      .execute(cancelToken: cancelToken);
-
-    final setEmailResponse = response.parse<SetEmailResponse>(
-      setEmailInvocation.methodCallId,
-      SetEmailResponse.deserialize
+    final emailCreated = await saveEmailAsDrafts(
+      session,
+      accountId,
+      newEmail,
+      cancelToken: cancelToken
     );
 
-    final emailUpdated = setEmailResponse?.created?[idCreateMethod];
-    final isEmailDeleted = setEmailResponse?.destroyed?.contains(oldEmailId.id);
-    final mapErrors = handleSetResponse([setEmailResponse]);
-
-    if (emailUpdated != null && isEmailDeleted == true && mapErrors.isEmpty) {
-      return emailUpdated;
-    } else {
-      throw SetMethodException(mapErrors);
+    try {
+      await removeEmailDrafts(
+        session,
+        accountId,
+        oldEmailId,
+        cancelToken: cancelToken
+      );
+    } catch (e) {
+      logError('EmailAPI::updateEmailDrafts: Exception = $e');
```

Comment:
IMO, we should combine in one jmap request, and only ignore the case of destroy error

Reply:
We cannot nest the email creation method. Because it will lead to a case of failed email creation but successful deletion. IMO, we should do it this way to ensure `delete email` is done only when `create email` succeeds.